ServerManager/classes/GameSave.php

```diff
@@ -178,7 +178,10 @@ public function createZip(GameSession $gameSession): bool
     {
         $urlBase = ServerManager::getInstance()->getAbsoluteUrlBase();
         if (getenv('DOCKER') !== false) {
-            $urlBase = 'http://caddy:80'.ServerManager::getInstance()->getAbsolutePathBase();
+            // this is always called from inside the docker environment,so just use http://caddy:80/ or
+            //   http://mitmproxy:8080/
+            $urlBase = 'http://'.($_ENV['WEB_SERVER_HOST'] ?? 'caddy').':'.($_ENV['WEB_SERVER_PORT'] ?? 80).
+                ServerManager::getInstance()->getAbsolutePathBase();
```

Comment:
please sift this out for a separate merge into dev so it can become part of 4.0.2